hazelcast/src/main/java/com/hazelcast/spi/impl/executionservice/impl/DelegatingTaskDecorator.java

```diff
@@ -45,6 +47,7 @@ public void run() {
         try {
             executor.execute(runnable);
         } catch (Throwable t) {
+            getLogger(getClass()).severe("Cannot execute " + runnable, t);
             ExceptionUtil.sneakyThrow(t);
```

Comment:
So is this sneakyThrow going to remain or not?

Imho the catch should be removed completely. The 'executor' should be configured with an exception logger.